### PR TITLE
Deprecate array util

### PR DESCRIPTION
**Pull request by hvy**
Raise deprecation warning on usage of `chainer.utils.array` since the functions included in the utility are all very primitive (and could in the future be removed).

May require discussion. Please refer to https://github.com/chainer/chainer/pull/3932.

### CODE REVIEW

File: tests/chainer_tests/links_tests/connection_tests/test_bilinear.py [hunk 1]
```diff
@@ -36,6 +35,12 @@ def _uniform(*shape):
     return numpy.random.uniform(-1, 1, shape).astype(numpy.float32)
 
 
+def _as_mat(x):
+    if x.ndim == 2:
```
Comment (okuta) [inline tests/chainer_tests/links_tests/connection_tests/test_bilinear.py]: Maybe, `x.ndim` is always `3`.

Comment (hvy) [reply tests/chainer_tests/links_tests/connection_tests/test_bilinear.py]: Thanks, I removed the condition.

**Comment by hvy** [comment]
Rebased to current master.

**Comment by okuta** [comment]
`./chainer/functions/connection/bilinear.py:6:1: F401 'array' imported but unused`

**Comment by hvy** [comment]
Thanks, I removed the unused import.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**